pkg/fanal/analyzer/language/analyze.go

```diff
@@ -90,6 +90,10 @@ func toApplication(fileType, filePath, libFilePath string, r dio.ReadSeekerAt, l
 
 	var pkgs []types.Package
 	for _, lib := range libs {
+		// At the moment `npm` only supports lib.Dev field
+		if !includeDevDeps && lib.Dev {
+			continue
+		}
```

Comment:
I think this filtering should be done when scanning. Manipulating analysis results affects the layer cache. We must discard the cache when the value of `--include-dev-deps` is changed. Rather than that, it is better to store devDependencies in the cache anyway., and then filter out them when scanning according to the CLI flag.

Reply:
You are absolutely right! Thank you for pointing out my omission.

I updated PR. Take a look, when you have time, please.

Reply:
Looks good! Thanks!